partial selecting of properties

rciurkot opened the pull request:
Tests method names (`testSelectPartial`) as well as method signatures (`fun(...).select(properties)`) suggest that using that method will cause only parts of object to be retrieved from db. However that's not the case in Kotlin support. Here `.select(properties)` always fetches full object. Also it is obligatory to provide in list of properties primary key of the table - otherwise crash will occur.

This implementation does what the name suggests - fetches only specified parts of object and returns it as a list of tuples.

npurushe commented:
Hmm this is working as intended currently. What case did it return the whole object for you? In the test by default the entity cache is enabled so the Person object is the same one as that was inserted, however if the cache is disabled a new partial object is returned with only the given properties.

rciurkot commented:
Ok, now I know that it was an intended behavior. I've reverted the changes and instead of removing original implementation I've left it untouched and added implementation using `Tuple`s.

`Tuple`-based approach is more "type"-safe as it will not produce degenerated object that have only some of the properties fetched.

npurushe commented:
There is an existing select method that takes expressions that returns a tuple, a method that limits the selection to just one type that doesn't return that type's object is probably overly narrow and doesn't match the java api. I think this can be easily added as an extension function in the calling application's code. In the future something like this accepting any list of properties can be part of requery with Kotlin 1.1's updated property reflection api.
